shared/js/ui/views/whitelist.es6.js

```diff
@@ -17,17 +20,87 @@ Whitelist.prototype = window.$.extend({},
   {
 
     _removeItem: function (e) {
-      var itemIndex = window.$(e.target).data('item')
+      const itemIndex = window.$(e.target).data('item')
+      const $clickedListItem = window.$(e.target).parent()
       this.model.removeDomain(itemIndex)
-      this.setWhitelistFromSettings()
+
+      // No need to rerender the whole view
+      // unless we need to show the "no sites in whitelist" message
+      if (this.$listitem && (this.$listitem.length > 1)) {
+        $clickedListItem.remove()
+      } else {
+        this.setWhitelistFromSettings()
+      }
```

Comment:
Still seeing some issues, with this, sorry. To reproduce:

1. Go to options.html with an empty whitelist
2. Add two items
3. Remove them both
4. You won't see the "no sites in whitelist" message

I was going to say "just check the model, it should have the latest" but it somehow doesn't...

Turns out the `this.$listitem.length` doesn't update

Reply:
Had a chat with @andrey-p and agreed on merging this as-is and filing this issue as a known bug.